src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/EntityAIWorkCowboy.java

```diff
@@ -0,0 +1,113 @@
+package com.minecolonies.coremod.entity.ai.citizen.herders;
+
+import com.minecolonies.coremod.colony.jobs.JobCowboy;
+import com.minecolonies.coremod.entity.ai.util.AIState;
+import com.minecolonies.coremod.entity.ai.util.AITarget;
+import net.minecraft.entity.passive.EntityCow;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.coremod.entity.ai.util.AIState.*;
+
+/**
+ * Created by Asher on 16/9/17.
+ */
+public class EntityAIWorkCowboy extends AbstractEntityAIHerder<JobCowboy, EntityCow>
+{
+    /**
+     * Max amount of animals per Hut Level.
+     */
+    private static final int MAX_ANIMALS_PER_LEVEL = 2;
+
+    /**
+     * Creates the abstract part of the AI.
+     * Always use this constructor!
+     *
+     * @param job the job to fulfill
+     */
+    public EntityAIWorkCowboy(@NotNull final JobCowboy job)
+    {
+        super(job, MAX_ANIMALS_PER_LEVEL);
+        itemsNiceToHave().add(new ItemStack(Items.BUCKET, 1));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "itemsNiceToHave" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)